### DISCUSSION TITLE
Using WinGet During MDT Task Sequence

### DISCUSSION THREAD

**Discussion by bftblomster** [microsoft/winget-cli#2462] [PowerShell]
A lot of our images do not have the Desktop App Installer app updated to include winget.exe

There is a way to trigger an update of all Store installed apps.  I wrote a script that does this and waits until version of the Desktop App Installer is 1.17 or higher. Once it is 1.17, then you can use winget.exe commands

```
Get-Ciminstance -Namespace "Root\cimv2\mdm\dmmap" -ClassName "MDM_EnterpriseModernAppManagement_AppManagement01" | Invoke-CimMethod -MethodName UpdateScanMethod
$MaxIterations = 12
$Iteration = 0
Do {
    $Iteration++
    Start-Sleep -Seconds 10
    If ($Iteration -eq $MaxIterations) {
        $Iteration = 0
        Get-Ciminstance -Namespace "Root\cimv2\mdm\dmmap" -ClassName "MDM_EnterpriseModernAppManagement_AppManagement01" | Invoke-CimMethod -MethodName UpdateScanMethod
    }
} Until ([System.Version](Get-AppxPackage -Name "Microsoft.DesktopAppInstaller").Version -ge [System.Version]"1.17")
```

**Comment by jantari** [comment]
Good find. If your deployment VLAN doesn't have unrestricted internet access or you don't want to wait for all apps to update you can also do an offline update using the msixbundle file from the release downloads:

https://github.com/microsoft/winget-cli/releases
